frontend/src/common/services/apis.ts

```diff
@@ -69,7 +69,16 @@ const baseQueryWithReAuth: BaseQueryFn<string | FetchArgs, unknown, FetchBaseQue
         const errorMessage = (result.error?.data as {detail: string})?.detail;
         if (errorMessage?.startsWith("CSRF Failed:")) {
             hdsToast.error("CSRF Virhe. Uudelleenohjataan kirjautumissivulle.");
-            window.location.href = getSignInUrl(window.location.href);
+            setTimeout(() => {
+                const isNotWriteOperation = ["GET", "HEAD", "OPTIONS"].includes(result.meta?.request.method ?? "");
+                if (isNotWriteOperation) {
+                    window.location.href = getSignInUrl(window.location.href);
+                } else {
```

Comment:
This part can be removed because CSRF error can't happen on a write operation.

Reply:
Fixed in https://github.com/City-of-Helsinki/hitas/compare/3aa5d19762513ba2561d4369dd2a31d674b770cb..33060648474bc3751f01a95cf0fab43deb240705